src/sawtooth/sawtooth.js

```diff
@@ -14,6 +14,152 @@ const log = require('../comm/util.js').log;
 let configPath;
 const request = require('request-promise');
 const TxStatus = require('../comm/transaction');
+const _ = require('lodash');
+const { Stream } = require('sawtooth-sdk/messaging/stream');
+const {
+    Message,
+    EventList,
+    EventSubscription,
+    ClientEventsSubscribeRequest,
+    ClientEventsSubscribeResponse,
+    ClientEventsUnsubscribeRequest,
+    ClientEventsUnsubscribeResponse
+} = require('sawtooth-sdk/protobuf');
+
+let lastKnownBlockId=null;
+let stream;
+let blockCommitSatus = new Map();
+let currentBlockNum=0;
+
+/**
+* Get the last recent block id for the block chain
+* @return {Promise<String>} last recent block id
+*/
+async function getCurrentBlockId() {
+    const request = require('request-promise');
+    let config = require(configPath);
+    let restApiUrl = config.sawtooth.network.restapi.url;
+    const blocks = restApiUrl + '/blocks?limit=1';
+    let options = {
+        uri: blocks
+    };
+    return request(options)
+        .then(function(body) {
+            let data = (JSON.parse(body)).data;
+            if (data.length > 0) {
+                currentBlockNum = parseInt(data[0].header.block_num);
+                lastKnownBlockId = data[0].header_signature.toString();
+                return currentBlockNum;
+            }
+        });
+}
+
+/**
+ * Get block data from event message
+ * @param {Object} events message
+ * @return {Promise<object>} The promise for the result of event message
+ */
+async function getBlock(events) {
+    const block = _.chain(events)
+        .find(e => e.eventType === 'sawtooth/block-commit')
+        .get('attributes')
+        .map(a => [a.key, a.value])
+        .fromPairs()
+        .value();
+    return {
+        blockNum: parseInt(block.block_num),
+        blockId: block.block_id.toString(),
+        stateRootHash: block.state_root_hash
+    };
+}
+
+/**
+ * Handle event message to updated lastKnownBlockId for next event subscription
+ * @param {Object} msg event message
+ * @return {void}
+ */
+async function handleEvent(msg) {
+    if (msg.messageType === Message.MessageType.CLIENT_EVENTS) {
+        const events = EventList.decode(msg.content).events;
+        getBlock(events).then(result => {
+            lastKnownBlockId = result.blockId.toString();
+            let blockNum=result.blockNum;
+            //On receiving event with block, update the status of the block to sucess
```

Comment:
sucess is a typo.

Reply:
After getting event from transactions commit event, updating the status to success.

Reply:
The last word 'sucess' of the annotation is a typo.

Reply:
It is success